lib/CArtHandler.cpp

```diff
@@ -758,20 +759,20 @@ void CArtHandler::erasePickedArt(const ArtifactID & id)
 		logMod->warn("Problem: cannot find list for artifact %s, strange class. (special?)", art->getNameTranslated());
 }
 
-boost::optional<std::vector<CArtifact*>&> CArtHandler::listFromClass( CArtifact::EartClass artifactClass )
+std::optional<std::reference_wrapper<std::vector<CArtifact *>>> CArtHandler::listFromClass(CArtifact::EartClass artifactClass)

```

Comment:
Perhaps remove this method and merge this code into `CArtHandler::erasePickedArt`?
This is the only one usage of this method and removal of optionals might actually clear up this code which got even more complex with this change

Reply:
done